docs/testing/workspaces-migration-guide.md

```diff
@@ -0,0 +1,319 @@
+        json!({
+            "force": true
+        })
+        .to_string()
+        .into_bytes(),
+        300_000_000_000_000 / 2,
+        1,
+    )
+    .submit();
+```
+
+There are no caveats here, the snippet can be straightforwardly mapped into the following:
+
+```rust title="Batch Transaction - workspace-rs"
+let res = contract
+    .batch(&worker)
+    .call(
+        Function::new("ft_transfer_call")
+            .args_json((defi_contract.id(), transfer_amount, Option::<String>::None, "10"))?
+            .gas(300_000_000_000_000 / 2)
+            .deposit(1),
+    )
+    .call(
+        Function::new("storage_unregister")
+            .args_json((Some(true),))?
+            .gas(300_000_000_000_000 / 2)
+            .deposit(1),
+    )
+    .transact()
+    .await?;
+```
+
+## Inspecting logs
+
+:::danger
+TODO: I found some potential inconsistencies in what I wrote about the log inspection behavior, need some time to look at this further
+:::
+
+The API for inspecting logs is fairly close to what it was in NEAR Simulator, but there are still some things you should keep in mind when migrating. Let's take the same transaction we used in the [batch transactions](#batch-transactions) section and try to inspect its logs. This is how one would check that the transaction logged a specific message in a certain position with NEAR Simulator:
+
+```rust title="Logs - near-sdk-sim"
+assert_eq!(
+    res.logs()[1],
+    format!("Closed @{} with {}", contract.account_id(), initial_balance - transfer_amount)
+);
+```
+
+The `workspaces-rs` counterpart might seem almost identical at the first look:
+
+```rust title="Logs - workspaces-rs"
+assert_eq!(
+    res.logs()[1],
+    format!("Closed @{} with {}", contract.id(), initial_balance.0 - transfer_amount.0)
+);
+```
+
+However, it can actually behave differently depending on your use case, because while near-sdk-sim version only returns the logs from the transaction, the workspaces version returns all logs from both the transaction and receipt outcomes. If you want a literal counterpart, please use `res.outcome().logs`.
+
+Another common use case is examining receipt outcome logs like this:
+
+```rust title="Logs - nead-sdk-sim"
+let outcome = res.get_receipt_results().remove(5).unwrap();
+
+assert_eq!(outcome.logs()[0], "The account of the sender was deleted");
+assert_eq!(
+    outcome.logs()[2],
+    format!("Account @{} burned {}", root.account_id(), 10)
+);
+```
+
+Which is straightforwardly replaced with:
+
+```rust title="Logs - workspaces-rs"
+let outcome = &res.receipt_outcomes()[5];
+assert_eq!(outcome.logs[0], "The account of the sender was deleted");
+assert_eq!(outcome.logs[2], format!("Account @{} burned {}", contract.id(), 10));
+```
+
+## Profiling gas
+
+NEAR Simulator has not been providing realistic gas estimates (TODO: needs elaboration - since when and why?), so the `workspaces-rs` gas estimates might come as a surprise. The good news is that they do reflect reality!
```

Comment:
@ChaoticTempest also probably yours :)

Reply:
> TODO: needs elaboration - since when and why?

it never had accurate gas estimations since it only tried to mirror nearcore, but nearcore has extra things on top which consumes gas (like cross-contract calls are processed separately from the same transaction and that incurs gas fees)

I'd give a warning that since workspaces-rs is now using accurate gas measurements, some contracts that were previously being tested with sdk-sim might not work the same, and that they should do their due diligence if they plan to deploy to mainnet. I mean, either ways, if they tried deploying the same contracts with infeasible gas estimates to testnet, it would have never worked. So we can say that workspaces offers the better experience here and aligns very well with what you can do on testnet and mainnet